# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/CanvasRenderingContext2D.cpp
# Commit: 5158484e9eab
# Full Hash: 5158484e9eabc55a48e73e68ad4b6e676d057997
# Author: Robert Longson <longsonr@gmail.com>
# Date: 2023-09-12 04:12:49
# Regressor Bug: 1287054
# File Overlap Count: 1
# Description:
#   Bug 1287054 part 2 - support vw, vh, vmin and vmax units for non-CSS lengths r=emilio,jgilbert
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D180619
# ==============================================================================

diff -r 35034282dc7c -r 5158484e9eab dom/canvas/CanvasRenderingContext2D.cpp
--- a/dom/canvas/CanvasRenderingContext2D.cpp	Mon Sep 11 18:20:18 2023 +0000
+++ b/dom/canvas/CanvasRenderingContext2D.cpp	Mon Sep 11 18:20:19 2023 +0000
@@ -2817,6 +2817,10 @@
   }
   gfx::Size GetSize() const override { return Size(mSize); }
 
+  CSSSize GetCSSViewportSize() const override {
+    return GetCSSViewportSizeFromContext(mPresContext);
+  }
+
  private:
   GeckoFontMetrics GetFontMetricsForType(Type aType) const override {
     switch (aType) {